### ISSUE TITLE
Crash report: IndexOutOfBoundsException (in CFAbstractAnalysis?)

### ISSUE THREAD

**Issue by kennknowles** [typetools/checker-framework#3793]
Steps to reproduce:

```
git clone --depth 1 --branch checker-curiosity https://github.com/kennknowles/beam
cd beam
./gradlew compileJava
```

Crash output:

```
error: SourceChecker.typeProcess: unexpected Throwable (IndexOutOfBoundsException) while processing /Users/klk/GitHub/apache/beam/runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableParDoViaKeyedWorkItems.java; message: Index: 0, Size: 0
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: /Users/klk/GitHub/apache/beam/runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableParDoViaKeyedWorkItems.java
  Last visited tree at line 71 column 1:
  public class SplittableParDoViaKeyedWorkItems {
  Exception: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0; java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
  	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
  	at java.util.ArrayList.get(ArrayList.java:433)
  	at java.util.Collections$UnmodifiableList.get(Collections.java:1311)
  	at org.checkerframework.framework.util.AnnotatedTypes.getAnnotatedTypes(AnnotatedTypes.java:949)
  	at org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism.resolve(AbstractQualifierPolymorphism.java:203)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.constructorFromUsePreSubstitution(GenericAnnotatedTypeFactory.java:1518)
  	at org.checkerframework.framework.type.AnnotatedTypeFactory.constructorFromUse(AnnotatedTypeFactory.java:2052)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.constructorFromUse(GenericAnnotatedTypeFactory.java:1507)
  	at org.checkerframework.framework.type.TypeFromExpressionVisitor.visitNewClass(TypeFromExpressionVisitor.java:320)
  	at org.checkerframework.framework.type.TypeFromExpressionVisitor.visitNewClass(TypeFromExpressionVisitor.java:70)
  	at com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1705)
  	at com.sun.source.util.SimpleTreeVisitor.visit(SimpleTreeVisitor.java:80)
  	at org.checkerframework.framework.type.TypeFromTree.fromExpression(TypeFromTree.java:38)
  	at org.checkerframework.framework.type.AnnotatedTypeFactory.fromExpression(AnnotatedTypeFactory.java:1288)
  	at org.checkerframework.framework.type.AnnotatedTypeFactory.getAnnotatedType(AnnotatedTypeFactory.java:1023)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.getValueFromFactory(CFAbstractTransfer.java:217)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitNode(CFAbstractTransfer.java:591)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitNode(CFAbstractTransfer.java:94)
  	at org.checkerframework.dataflow.cfg.node.AbstractNodeVisitor.visitObjectCreation(AbstractNodeVisitor.java:333)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitObjectCreation(CFAbstractTransfer.java:962)
  	at org.checkerframework.framework.flow.CFAbstractTransfer.visitObjectCreation(CFAbstractTransfer.java:94)
  	at org.checkerframework.dataflow.cfg.node.ObjectCreationNode.accept(ObjectCreationNode.java:63)
  	at org.checkerframework.dataflow.analysis.AbstractAnalysis.callTransferFunction(AbstractAnalysis.java:338)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.callTransferFunction(ForwardAnalysisImpl.java:375)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysisBlock(ForwardAnalysisImpl.java:152)
  	at org.checkerframework.dataflow.analysis.ForwardAnalysisImpl.performAnalysis(ForwardAnalysisImpl.java:106)
  	at org.checkerframework.framework.flow.CFAbstractAnalysis.performAnalysis(CFAbstractAnalysis.java:109)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1296)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:1204)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.checkAndPerformFlowAnalysis(GenericAnnotatedTypeFactory.java:1615)
  	at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.preProcessClassTree(GenericAnnotatedTypeFactory.java:293)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:318)
  	at org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:171)
  	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
  	at org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
  	at org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:916)
  	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:502)
  	at org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:190)
```

**Comment by mernst** [comment]
See comment in #3794.

**Comment by smillst** [comment]
This is a duplicate of #3776.  (I can't reproduce this using a version of the Checker Framework with the fix for #3776.)
